packages/debug/src/browser/view/debug-configuration-select.tsx

```diff
@@ -0,0 +1,223 @@
+/********************************************************************************
+ * Copyright (C) 2021 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import URI from '@theia/core/lib/common/uri';
+import * as React from '@theia/core/shared/react';
+import { DebugConfigurationManager } from '../debug-configuration-manager';
+import { DebugSessionOptions, InternalDebugSessionOptions } from '../debug-session-options';
+import { QuickInputService } from '@theia/core/lib/browser';
+import { nls } from '@theia/core/lib/common/nls';
+
+interface DynamicPickItem { label: string, configurationType: string, request: string, providerType: string }
+
+export interface DebugConfigurationSelectProps {
+    manager: DebugConfigurationManager,
+    quickInputService: QuickInputService,
+    isMultiRoot: boolean
+}
+
+export interface DebugProviderSelectState {
+    providerTypes: string[]
+}
+
+export class DebugConfigurationSelect extends React.Component<DebugConfigurationSelectProps, DebugProviderSelectState> {
```

Comment:
I don't think that it is very likely that this particular element will be overridden downstream, but in principle an injectable `ReactRenderer` would make the task easier for adopters.

Reply:
As per our discussion, lets keep this in mind for a future enhancement, when needed !
Thanks !!